packages/deployer/internal/storage-ast-validator.js

```diff
@@ -139,15 +139,17 @@ class ModuleStorageASTValidator {
     const previousStructsMap = await buildContractsStructMap(this.previousAsts);
     const currentStructsMap = await buildContractsStructMap(this.asts);
 
-    const { modifications, removals } = compareStorageStructs({
+    let { modifications, removals } = compareStorageStructs({
       previousStructsMap,
       currentStructsMap,
     });
 
+    removals = removals.filter((removal) => removal.completeStruct === false);
```

Comment:
Why?

This change makes this function `findInvalidNamespaceMutations()` only about the internals of a struct, ignoring complete struct changes.

The idea is that other function `findNamespaceSlotChanges()` focuses on slots as a whole.